DataFormats/CSCDigi/interface/CSCALCTDigi.h

```diff
@@ -35,50 +39,56 @@ class CSCALCTDigi {
   void setValid(const int valid) { valid_ = valid; }
 
   /// return quality of a pattern
-  int getQuality() const { return quality_; }
+  uint16_t getQuality() const { return quality_; }
 
   /// set quality
   void setQuality(const int quality) { quality_ = quality; }
 
   /// return Accelerator bit
   /// 1-Accelerator pattern, 0-CollisionA or CollisionB pattern
-  int getAccelerator() const { return accel_; }
+  uint16_t getAccelerator() const { return accel_; }
 
   /// set accelerator bit
   void setAccelerator(const int accelerator) { accel_ = accelerator; }
 
   /// return Collision Pattern B bit
   /// 1-CollisionB pattern (accel_ = 0),
   /// 0-CollisionA pattern (accel_ = 0)
-  int getCollisionB() const { return patternb_; }
+  uint16_t getCollisionB() const { return patternb_; }
 
   /// set Collision Pattern B bit
   void setCollisionB(const int collision) { patternb_ = collision; }
 
   /// return key wire group
-  int getKeyWG() const { return keywire_; }
+  uint16_t getKeyWG() const { return keywire_; }
 
   /// set key wire group
   void setKeyWG(const int keyWG) { keywire_ = keyWG; }
 
   /// return BX - five low bits of BXN counter tagged by the ALCT
-  int getBX() const { return bx_; }
+  uint16_t getBX() const { return bx_; }
 
   /// set BX
   void setBX(const int BX) { bx_ = BX; }
 
   /// return track number (1,2)
-  int getTrknmb() const { return trknmb_; }
+  uint16_t getTrknmb() const { return trknmb_; }
 
   /// Set track number (1,2) after sorting ALCTs.
   void setTrknmb(const uint16_t number) { trknmb_ = number; }
 
   /// return 12-bit full BX.
-  int getFullBX() const { return fullbx_; }
+  uint16_t getFullBX() const { return fullbx_; }
 
   /// Set 12-bit full BX.
   void setFullBX(const uint16_t fullbx) { fullbx_ = fullbx; }
 
+  /// return the high multiplicity bits
+  uint16_t getHMT() const { return ((version_ == Version::Run3) ? hmt_ : -1); }
```

Comment:
Probably not a great idea to return -1 as an unsigned int; maybe better to use 0?  (Same for the setHMT below.)

Reply:
Here, 0 means "no high multiplicity event in chamber", which makes sense for Run-3, but not for legacy data since the HMT was not enabled then. Hence the "-1" for legacy

Reply:
Right, but the function will not return "-1", it will return some large positive number (I think 2^16 - 1, so 65535), since the function returns a uint16_t.  Maybe this is OK, but 65535 is a less obvious "Legacy / Invalid" flag than 0, or 999, or something like that.

Reply:
"it will return some large positive number" - Exactly. This is what I had in mind.

Reply:
`std::numeric_limits<uint16_t>::max()` would be a bit more clear about that.

Reply:
I would not use -1, @makortel proposal is the most classical

Reply:
> Here, 0 means "no high multiplicity event in chamber", which makes sense for Run-3, but not for legacy data since the HMT was not enabled then.

So, in a sense, there could not be any high multiplicity events, and 0 could make sense ?
But I guess it depends on how he value will be used in the client code.

Reply:
Yes, it is very likely that a chamber had 1 or 2 LCTs, without any high multiplicity hits (extra LCTs, or an unusual large number of single hits). In that case HMT = 0.